ldms/src/sampler/procnetdev2/procnetdev2.c

```diff
@@ -417,38 +481,67 @@ static int sample(struct ldmsd_sampler *self)
 
 static void term(struct ldmsd_plugin *self)
 {
-	if (mf)
-		fclose(mf);
-	mf = NULL;
-	base_set_delete(base);
-	base_del(base);
-	base = NULL;
-	if (mylog)
-		ovis_log_destroy(mylog);
+	procnetdev2_t p = (void*)self;
+	if (p->mf) {
+		fclose(p->mf);
+		p->mf = NULL;
+	}
+	p->mf = NULL;
+	base_set_delete(p->base_data);
+	base_del(p->base_data);
+	p->base_data = NULL;
+
+	if (p->iface) {
+		free(p->iface);
+		p->iface = NULL;
+	}
+	if (p->iface_str) {
+		free(p->iface_str);
+		p->iface_str = NULL;
+	}
+	p->niface = 0;
 }
 
 
-static struct ldmsd_sampler procnetdev2_plugin = {
-	.base = {
-		.name = SAMP,
-		.type = LDMSD_PLUGIN_SAMPLER,
-		.term = term,
-		.config = config,
-		.usage = usage,
-	},
-	.get_set = get_set,
-	.sample = sample,
-};
+static void __procnetdev2_del(struct ldmsd_cfgobj *self)
+{
+	procnetdev2_t p = (void*)self;
+	free(p);
+}
 
-struct ldmsd_plugin *get_plugin()
+static void __once()
```

Comment:
The existence of this __once() function tells us that there is a problem with the API design. No plugin author should need to do this. We are failing to give the author a place to do global initialization (and also missing the matching place to tear down global state at the end) that will be used, regardless of how many instances there may be.

We could fix this by going with my suggestion (or something similar) of having ldmsd_import()/ldmsd_export() calls (that are independant of the instance_init()/instance_fini() methods).